### PR TITLE
Remove PHP5.2 code

### PR DESCRIPTION
**Pull request by Hywan**
Noticing this PHP5.2 code. Now we dropped it, we can clean the code.

### CODE REVIEW

File: lib/DAVACL/Plugin.php [hunk 1]
```diff
@@ -416,47 +416,44 @@ static function getDefaultSupportedPrivilegeSet() {
...
+
+                }
 
             }
 
-        }
+        };
+
+        $flat = [];
+        $fpmsTraverse($privs, null, $flat);
```
Comment (staabm) [inline lib/DAVACL/Plugin.php]: should this be `$fpsTraverse`?

**Comment by evert** [comment]
Awesome =) this was indeed quite old school.

**Comment by Hywan** [comment]
I used an old trick for the recursivity. Don't know if you have something better :-/.

**Comment by evert** [comment]
That's also how i normally do that, You could use a y-combinator but it's overkill :P

**Comment by Hywan** [comment]
How?

**Comment by evert** [comment]
It's mostly a joke, I think this is the best way =)

**Comment by evert** [comment]
https://php100.wordpress.com/2009/04/13/php-y-combinator/

**Comment by Hywan** [comment]
Oh, this one: https://en.wikipedia.org/wiki/Fixed-point_combinator#Y_combinator, I though http://www.ycombinator.com/ ;-).

**Pull request merged**